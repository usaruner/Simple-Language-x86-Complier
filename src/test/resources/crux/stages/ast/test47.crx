// scoping test
var a:int;

func main() : void {
  let a = 1;
  if (true) {
    var a:int;
    let a = 2;
    if (true) {
      let a = a * 10;
      var a:int;
      let a = 3;
      if(true) {
        var a:bool;
	let a = true;
	::printBool(a);
	::println();
      }
      ::printInt(a);
      ::println();
    }
    ::printInt(a);
    ::println();
  }
  ::printInt(a);
  ::println();
}
  
