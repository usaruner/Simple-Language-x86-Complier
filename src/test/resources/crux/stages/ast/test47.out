ast.DeclarationList(2)
  ast.VariableDeclaration(2)[Symbol(a)]
  ast.FunctionDefinition(4)[Symbol(main), []]
    ast.StatementList(5)
      ast.Assignment(5)
        ast.Name(5)[a]
        ast.LiteralInt(5)[1]
      ast.IfElseBranch(6)
        ast.LiteralBool(6)[TRUE]
        ast.StatementList(7)
          ast.VariableDeclaration(7)[Symbol(a)]
          ast.Assignment(8)
            ast.Name(8)[a]
            ast.LiteralInt(8)[2]
          ast.IfElseBranch(9)
            ast.LiteralBool(9)[TRUE]
            ast.StatementList(10)
              ast.Assignment(10)
                ast.Name(10)[a]
                ast.OpExpr(10)[MULT]
                  ast.Dereference(10)
                    ast.Name(10)[a]
                  ast.LiteralInt(10)[10]
              ast.VariableDeclaration(11)[Symbol(a)]
              ast.Assignment(12)
                ast.Name(12)[a]
                ast.LiteralInt(12)[3]
              ast.IfElseBranch(13)
                ast.LiteralBool(13)[TRUE]
                ast.StatementList(14)
                  ast.VariableDeclaration(14)[Symbol(a)]
                  ast.Assignment(15)
                    ast.Name(15)[a]
                    ast.LiteralBool(15)[TRUE]
                  ast.Call(16)[Symbol(printBool)]
                    ast.Dereference(16)
                      ast.Name(16)[a]
                  ast.Call(17)[Symbol(println)]
                ast.StatementList(13)
              ast.Call(19)[Symbol(printInt)]
                ast.Dereference(19)
                  ast.Name(19)[a]
              ast.Call(20)[Symbol(println)]
            ast.StatementList(9)
          ast.Call(22)[Symbol(printInt)]
            ast.Dereference(22)
              ast.Name(22)[a]
          ast.Call(23)[Symbol(println)]
        ast.StatementList(6)
      ast.Call(25)[Symbol(printInt)]
        ast.Dereference(25)
          ast.Name(25)[a]
      ast.Call(26)[Symbol(println)]
